pylot/simulation/perfect_lane_detector_operator.py

```diff
@@ -0,0 +1,90 @@
+from erdos.op import Op
+from erdos.utils import setup_csv_logging, setup_logging
+
+from pylot.utils import create_detected_lane_stream, is_can_bus_stream
+from pylot.simulation.utils import DetectedLane, to_pylot_location
+from pylot.simulation.messages import DetectedLaneMessage
+from pylot.simulation.carla_utils import get_world, to_carla_location
+
+
+class PerfectLaneDetectionOperator(Op):
+    def __init__(self,
+                 name,
+                 output_stream_name,
+                 flags,
+                 log_file_name=None,
+                 csv_file_name=None):
+        """ Initializes the PerfectLaneDetectionOperator to use the given name
+        and output to the given stream.
+
+        Args:
+            name: The name to be given to the operator.
+            output_stream_name: The name of the output stream.
+            flags: The flags to be used while initializing.
+            log_file_name: Name of the log file.
+            csv_file_name: Name of the csv file.
+        """
+        super(PerfectLaneDetectionOperator, self).__init__(name)
+        self._flags = flags
+        self._logger = setup_logging(self.name, log_file_name)
+        self._csv_logger = setup_csv_logging(self.name + '-csv', csv_file_name)
+        self._output_stream_name = output_stream_name
+
+        _, world = get_world(self._flags.carla_host, self._flags.carla_port,
```

Comment:
Can you move this logic into execute? You might end up getting a stale map if you have this logic here when using Carla 0.9.6. The carla operator reloads the map in the init method to set it to the desired town. However, all the init methods are called simultaneously by ERDOS, and thus this get_world invocation might execute before carla operator reloads the map, and the server applies.